src/python/module/smartredis/util.py

```diff
@@ -111,3 +114,93 @@ def typecheck(arg, name, _type):
     """
     if not isinstance(arg, _type):
         raise TypeError(f"Argument {name} is of type {type(arg)}, not {_type}")
+
+
+
+class DatasetConverter:   
+    
+    def add_metadata_for_xarray(ds,data=None,dims=None,coords=None,attrs=None): #time=None,time_variable=None):
+        """_summary_
+        return attribute labels, dimensions labels, coordinate labels etc for retrieval 
+        adding the field names to retrieve to xarray specific field names 
+        can use those general field names to construct xarray in transform method 
+
+        Args:
+            ds (_type_): _description_
+            data (_type_, optional): _description_. Defaults to None.
+            dims (_type_, optional): _description_. Defaults to None.
+            coords (_type_, optional): _description_. Defaults to None.
+            attrs (_type_, optional): _description_. Defaults to None.
+        """
+        
+        args = list(locals().keys())[1:]
+        for arg in args: 
+            if isinstance(eval(arg),list):
+                items = []
+                for item in eval(arg): 
+                    items.append(item)
+        
+                arg_field=",".join(items)
+             
+                ds.add_meta_string(f"_xarray_{arg}",arg_field)  # pass in comma separated string 
+            else:
+                if eval(arg) != None: 
+                    ds.add_meta_string(f"_xarray_{arg}",eval(arg))
+                    
+
+     
+    def transform_to_xarray(ds): 
+        """_summary_
+
+        Args:
+            ds (_type_): _description_
+
+        Returns:
+            xarray.DataArray: _description_
+        """
+    
+        # data 
+        try: 
+            for item in ds.get_meta_strings("_xarray_data")[0].split(","):    # for the data field if list - prepare to retrieve more than one tensor 
+                _xarray_data = ds.get_tensor(f"{item}")
```

Comment:
Does this actually work for more than one tensor? Looks like you're overwriting `_xarray_data` here if there are more than one

Reply:
I think the next line should read `data_final.append(_xarray_data)`?